File: 1423.py
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        total = sum(cardPoints)
        length = len(cardPoints) - k
        current = sum(cardPoints[:length])
        min_subarray = current

        for i in range(length, len(cardPoints)):
            current += cardPoints[i] - cardPoints[i - length]
            min_subarray = min(min_subarray, current)
        
        return total - min_subarray

File: 1423.java
class Solution {
    public int maxScore(int[] cardPoints, int k) {
        int total = 0;
        for (int num : cardPoints) {
            total += num;
        }

        int len = cardPoints.length - k;
        if (len == 0) return total; 

        int current = 0;
        for (int i = 0; i < len; i++) {
            current += cardPoints[i];
        }

        int minSubarraySum = current;
        for (int i = len; i < cardPoints.length; i++) {
            current += cardPoints[i] - cardPoints[i - len];
            minSubarraySum = Math.min(minSubarraySum, current);
        }

        return total - minSubarraySum;
    }
}
